services/user/Supervisor/ui/src/plugin/pluginHost.ts

```diff
@@ -215,16 +227,58 @@ export class PluginHost implements HostInterface {
     getAppUrl(app: string): string {
         return `${siblingUrl(null, app)}`;
     }
+    
+    private setActiveUserPrompt(subpath: string | undefined, upPayloadJsonStr: string): Result<string, RecoverableErrorPayload> {
+        let up_id = window.crypto.randomUUID?.() ?? Math.random().toString();
+        
+        // In Supervisor localStorage space, store id, subdomain, and subpath
+        const supervisorUP = SupervisorPromptPayload.parse({
+            id: up_id,
+            subdomain: this.self.app,
+            subpath: subpath,
+            expiry_timestamp: Math.floor(new Date().getTime() / 1000),
```

Comment:
If the other comment is followed 
```suggestion
            expiry_timestamp: Math.floor(new Date().getTime() / 1000) + OAUTH_REQUEST_EXPIRATION,
```